### PR TITLE
Refactor/all refactoring

### PR DESCRIPTION
**Pull request by denissosnowsky**

### CODE REVIEW

File: src/components/common/button-icon/button-icon.tsx [hunk 1]
```diff
@@ -1,49 +1,56 @@
...
+    <>
+      {hasOpacity ? (
+        <TouchableOpacity onPress={onPress} accessibilityRole="button" style={styles.wrapper}>
+          {Icon}
+        </TouchableOpacity>
+      ) : (
+        <TouchableWithoutFeedback
+          onPress={onPress}
+          accessibilityRole="button"
+          style={styles.wrapper}
```
Comment (ivnmarkov) [inline src/components/common/button-icon/button-icon.tsx:40]: Same props in both components can be moved to one object and then put into both of them
{...componentsProps}

File: src/components/common/list-item/components/list-item-status/list-item-status.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  isTodoSelectedOrCompleted,
+  isMultipleEditingMode,
+  onSelectTodoHandler,
+  onCheckPressTodoHandle,
+}) => {
+  return (
+    <View style={styles.complete}>
+      {isTodoSolelyEditing ? (
+        <ButtonIcon testID="pencil" variant="pencil" size={globalStyles.ICON_EXSM_SIZE} />
+      ) : isTodoSelectedOrCompleted ? (
```
Comment (ivnmarkov) [inline src/components/common/list-item/components/list-item-status/list-item-status.tsx:19]: Might be better to move to a separate function that return component

**Comment by ivnmarkov** [comment]
Great job! The code is much cleaner now. Keep up the good work😊👍

File: src/components/common/button-icon/button-icon.tsx (hunk 1 above)
Comment (denissosnowsky) [reply src/components/common/button-icon/button-icon.tsx:40]: Moved to one object named 'componentsProps'

File: src/components/common/list-item/components/list-item-status/list-item-status.tsx (hunk 2 above)
Comment (denissosnowsky) [reply src/components/common/list-item/components/list-item-status/list-item-status.tsx:19]: Moved 'else' part to separate function

**Pull request merged**